front-end/src/App.js

```diff
@@ -12,6 +14,8 @@ function App() {
         <Switch>
           <Route exact path="/" component={Gallery}/>
           <Route exact path="/add" component={AddProduct}/>
+          <Route exact path="/:id" component={ProductView}/>
+          <Route exact path="/edit/:id" component={EditProduct} />
```

Comment:
Should this line be before `<Route exact path="/:id" component={ProductView}/>`, because it will never be fired, I think?

Reply:
It works but I can switch them.

Reply:
Weird, but okay if it works don't touch it. 🙈